### PR TITLE
Add support to filter errors in packages based on codes

### PR DESCRIPTION
**Pull request by mrodm**
Relates https://github.com/elastic/package-spec/issues/558

This PR adds the required support to filter errors based on the errors configuration (~`errors.yml`~ `validation.yml`) for both directories and zip packages.

This PR depends on a new package-spec version including the structured errors.

### CODE REVIEW

**Comment by mrodm** [comment]
buildkite test this

**Comment by mrodm** [comment]
buildkite test this

File: internal/validation/validation.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		return allErrors, nil, nil
+	}
+
+	_, err := fs.Stat(fsys, configPath)
+	if err != nil {
+		logger.Debugf("file not found: %s", configPath)
+		return allErrors, nil, nil
+	}
+
+	config, err := processors.LoadConfigFilter(fsys, configPath)
```
Comment (jsoriano) [inline internal/validation/validation.go:96]: I still see this "processors" name as a too generic name, specially when used externally, but lets continue with it.

**Review by jlind23 — APPROVED** [review]
LGTM

**Review by jsoriano — APPROVED** [review]
Ship it! :boat:

**Pull request merged**